### ISSUE TITLE
A file that is locked in the WebGUI will fail to (renew?) lock via WebDAV

### ISSUE THREAD

**Issue by carlolars** [danmunn/redmine_dmsf#617]
When working with MsOffice then opening a document for editing via the webdav share also locks the document, and it is unlocked when the document is closed.
But if the document is already locked in the WebGUI then when opening the file from the WebDAV share with the same user Office will fail to lock the file and not be able to open the file.
Even Notepad.exe locks the file when saving and unlocks directly after the save is done (but why???), and that lock also fails if the file is already locked in the WebGUI by the same user that tried to open it in the WebDAV.

The expected behavior would be that if you lock a file in the WebGUI then lock requests via the WebDAV should be successful (renew lock).

**Comment by picman** [comment]
I've studied it for a while... If we lock a document from the GUI, the default lock type is _exclusive_. It means that nobody else is allowed to lock the document. MS Word and MS Notepad as I've tested tried to make an exclusive lock on the document:
```
Notepad: INFO -- : >>> {:depth=>:infinity, :scope=>"exclusive", :type=>"write", :owner=>"kpicman"}
Word   : INFO -- : >>> {:depth=>:infinity, :scope=>"exclusive", :type=>"write", :owner=>"kpicman"}
```
Such attempt to create an exclusive lock on an already locked document certainly fails.

I'd consider this behaviour as the correct one.

If I tried other editors such as LiberOffice or gedit, they do not lock edited documents and simple send GET and PUT commands which succeed if the edited document had been lock by the same user as it's being edited.
